core/templates/dev/head/services/contextual/WindowDimensionsService.ts

```diff
@@ -16,26 +16,37 @@
  * @fileoverview Service for computing the window dimensions.
  */
 
-angular.module('oppia').factory('WindowDimensionsService', [
-  '$window', function($window) {
-    var onResizeHooks = [];
-    angular.element($window).bind('resize', function() {
-      onResizeHooks.forEach(function(hookFn) {
-        hookFn();
-      });
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { HostListener, Injectable } from '@angular/core';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class WindowDimensionsService {
+  onResizeHooks : Array<any> = [];
+
+  @HostListener('window:resize', ['$event'])
+  onResize(event) {
+    this.onResizeHooks.forEach(function(hookFn) {
+      hookFn();
     });
-    return {
-      getWidth: function() {
-        return (
-          $window.innerWidth || document.documentElement.clientWidth ||
-          document.body.clientWidth);
-      },
-      registerOnResizeHook: function(hookFn) {
-        onResizeHooks.push(hookFn);
-      },
-      isWindowNarrow: function() {
-        var NORMAL_NAVBAR_CUTOFF_WIDTH_PX = 768;
-        return this.getWidth() <= NORMAL_NAVBAR_CUTOFF_WIDTH_PX;
-      }
-    };
-  }]);
+  }
+
+  getWidth(): number {
+    return ( window.innerWidth || document.documentElement.clientWidth ||
+        document.body.clientWidth);
+  }
+
+  registerOnResizeHook(hookFn): void {
```

Comment:
Type of argument?

Reply:
I am not sure what's the type of this..can i use any keyword?

Reply:
So, you are pushing a object into an Array that is of type any. So, it should be any for now. Check line 26.

Reply:
Done